### PR TITLE
JDK-8324337: Cherry-pick WebKit 617.1 stabilization fixes

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Cherry-picked changes related to webkit-2.42.4.Verified build on all platforms. Sanity testing looks fine.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8324337](https://bugs.openjdk.org/browse/JDK-8324337): Cherry-pick WebKit 617.1 stabilization fixes (**Bug** - P3)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Joeri Sykora](https://openjdk.org/census#sykora) (@tiainen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1350/head:pull/1350` \
`$ git checkout pull/1350`

Update a local copy of the PR: \
`$ git checkout pull/1350` \
`$ git pull https://git.openjdk.org/jfx.git pull/1350/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1350`

View PR using the GUI difftool: \
`$ git pr show -t 1350`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1350.diff">https://git.openjdk.org/jfx/pull/1350.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1350#issuecomment-1910238338)

### CODE REVIEW

**Review by kevinrushforth — COMMENTED** [review]
Code changes look fine with one suggestion in the added code under `#if PLATFOMR(JAVA)`.

I'll test it and finish my review.

File: modules/javafx.web/src/main/native/Source/WebCore/rendering/updating/RenderTreeBuilderMultiColumn.cpp [hunk 1]
```diff
@@ -187,10 +187,15 @@ void RenderTreeBuilder::MultiColumn::restoreColumnSpannersForContainer(const Ren
...
+#if PLATFORM(JAVA)
+       if(spannerAndPlaceholder.value.get() != nullptr)
+#endif
+        {
+            if (!placeholder.isDescendantOf(&container))
             continue;
-        placeholdersToRestore.append(&placeholder);
-    }
+            placeholdersToRestore.append(&placeholder);
+        }
```
[the remark this answers is not in the document]
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/WebCore/rendering/updating/RenderTreeBuilderMultiColumn.cpp]: done

**Review by kevinrushforth — APPROVED** [review]
Looks good now.

Minor: the indentation is still a bit off, but I'll approve as-is. If you decide to fix it, I'll reapprove.

**Comment by kevinrushforth** [comment]
@tiainen or @johanvos can one of you be the second reviewer?

**Review by kevinrushforth — APPROVED** [review]

**Review by tiainen — APPROVED** [review]
Everything built fine.

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
@tiainen Can you rereview? (only whitespace changed from last time)

**Review by tiainen — APPROVED** [review]

**Pull request closed without merging**